components/profile-controller/controllers/profile_controller.go

```diff
@@ -616,15 +624,59 @@ func removeString(slice []string, s string) (result []string) {
 	return
 }
 
+func mergeLabelPropertiesToMap(logger logr.Logger) {
+	file, err := os.Open(NAMESPACE_LABELS_PROPERTIES)
+	if err != nil {
+		logger.Info("namespace labels properties file doesn't exist, using default value")
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		label := scanner.Text()
+		arr := strings.Split(label, "=")
+		if len(arr) > 2 || len(arr) == 0 {
+			logger.Info("label config format incorrect, should be {key}={value}", "label", label)
+		} else {
+			if arr[0] == "" {
+				logger.Info("label key should not be empty", "label", label)
+			} else if len(arr) == 2 {
+				// Add or overwrite label map if value exists.
+				kubeflowNamespaceLabels[arr[0]] = arr[1]
+			} else {
+				// Set value to be empty if nothing exists after "=".
+				// It means this label needs to be removed.
+				kubeflowNamespaceLabels[arr[0]] = ""
+			}
+		}
+	}
```

Comment:
It's a nice implementation of the syntax you are proposing.

However, it seems a little unnecessary if we can just use YAML format as the config format, so people do not need to reinvent the wheels and consumers won't guess about how the format is parsed. There are standard libraries for parsing YAML in go: https://github.com/go-yaml/yaml/tree/v2.

Using the standards mean that e.g. it's very easy to insert comments in the configfile, without you needing to implement logic to ignore comments.

Reply:
Thank you for the suggestion! I have switched to yaml format in new commit.